Lombiq.BaseTheme/Services/CssClassHolder.cs

```diff
@@ -21,5 +24,15 @@ public void AddClassToZone(string zoneName, string className)
         classes.Add(className);
     }
 
-    public ISet<string> GetZoneClasses(string zoneName) => _zones.GetMaybe(zoneName) ?? new();
+    [Obsolete($"Use {nameof(GetOrAddZoneClasses)} instead.")]
+    public ISet<string> GetZoneClasses(string zoneName) => GetOrAddZoneClasses(zoneName);
+
+    public ISet<string> GetOrAddZoneClasses(string zoneName)
+    {
+        if (_zones.TryGetValue(zoneName, out var zone)) return zone;
+
+        zone = new HashSet<string>();
+        _zones[zoneName] = zone;
+        return zone;
```

Comment:
I feel that's a lot of code for the simple assignment and return. Do you think this is too unreadable?

```suggestion
        return (_zones[zoneName] = new HashSet<string>());
```

Reply:
[The analyzer doesn't permit it](https://rules.sonarsource.com/csharp/RSPEC-1121) and I completely agree. It's hard to reason about assignments inside an expression, since you have to think about evaluation order and the like. It's best for C# to treat assignments as if they were statements instead of expressions.